pkg/cookies/cookies.go

```diff
@@ -7,14 +7,34 @@ import (
 	"strings"
 	"time"
 
-	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/options"
 	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/logger"
 	requestutil "github.com/oauth2-proxy/oauth2-proxy/v7/pkg/requests/util"
 )
 
+type CookieOptions struct {
+	// Cookie name
+	Name string
+	// Cookie value
+	Value string
+	// Domains
+	Domains []string
+	// Cookie expiration duration
+	Expiration time.Duration
+	// current time
+	Now time.Time
+	// Samesite
+	SameSite string
+	// Cookie path
+	Path string
+	// HTTP only
+	HTTPOnly bool
+	// Secure
+	Secure bool
+}
+
 // MakeCookieFromOptions constructs a cookie based on the given *options.CookieOptions,
 // value and creation time
-func MakeCookieFromOptions(req *http.Request, name string, value string, opts *options.Cookie, expiration time.Duration, now time.Time) *http.Cookie {
+func MakeCookieFromOptions(req *http.Request, opts *CookieOptions) *http.Cookie {
```

Comment:
See errors reported below. expiration and now aren't defined anymore but still used.